etl/src/energuide/transform.py

```diff
@@ -89,8 +109,10 @@ def _generate_dwellings(grouped: typing.List[typing.Dict[str, typing.Any]]) -> t
     return None
 
 
-def transform(extract_reader: ExtractProtocol) -> typing.Iterator[dwelling.Dwelling]:
-    for row_group in _read_groups(extract_reader.extracted_rows()):
+def transform(extract_reader: ExtractProtocol, show_progress: bool = False) -> typing.Iterator[dwelling.Dwelling]:
```

Comment:
Also, all tests that iterated through the `tqdm`-ified iterator caused console output, particularly when the tests failed. That made for some messy logs to parse through, and in one test where we were explicitly testing that a specific error message was logged, it caused that test to fail (since there was unrelated unexpected output to stderr).